File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        
        closed_to_open = {')': '(', '}': '{', ']': '['}

        for c in s:
            if c not in closed_to_open:
                stack.append(c)
            else:
                if (len(stack) == 0 or stack.pop() != closed_to_open[c]):
                    return False
        
        return len(stack) == 0

File: Data Structures & Algorithms/daily-temperatures/submission-3.kt
class Solution {
    fun dailyTemperatures(temperatures: IntArray): IntArray {
        val stack = ArrayDeque<Int>()
        val res = MutableList(temperatures.size) {0}

        for (i in 0..temperatures.size-1) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.last()]) {
                
                val prev_idx = stack.removeLast()
                res[prev_idx] = i - prev_idx
            }
            stack.addLast(i)
        }
        return res.toIntArray()
    }
}
